prov/usnic/src/usdf_domain.c

```diff
@@ -274,16 +278,30 @@ usdf_domain_open(struct fid_fabric *fabric, struct fi_info *info,
 	switch (info->addr_format) {
 	case FI_SOCKADDR:
 		addrlen = sizeof(struct sockaddr);
+		sin = info->src_addr;
 		break;
 	case FI_SOCKADDR_IN:
 		addrlen = sizeof(struct sockaddr_in);
+		sin = info->src_addr;
+		break;
+	case FI_ADDR_STR:
+		USDF_DBG("info->src_addr: %s\n", info->src_addr);
+		usdf_str_toaddr(info->src_addr, &tmp_sin);
+		src = *tmp_sin;
+		sin = &src;
+		free(tmp_sin);
+		goto skip_size_check;
```

Comment:
instead of using `tmp_sin` & `src`here, couldn't you just use `sin` here and free it at the end of the function?